src/components/CreateAccount.js

```diff
@@ -90,14 +90,14 @@ class CreateAccount extends Component {
   }
 
   componentDidUpdate() {
-    const { step, query } = this.state;
+    const { step, query, username } = this.state;
     if (step === 'created') {
       if (this.isWhistle()) {
         window.postMessage('whistle_signup_complete');
       } else {
         const urlParameters = query && Object.keys(query).map(param => `${param}=${query[param]}`).join('&');
         setTimeout(() => {
-          window.location.href = `${window.config.DEFAULT_REDIRECT_URI}?${urlParameters}`;
+          window.location.href = `${window.config.DEFAULT_REDIRECT_URI}?user=${username}&${urlParameters}`;
```

Comment:
This is hardcoded to `?user`. It should be configurable

Reply:
If I add a `REDIRECT_USER_PARAMETER` is it OK ? If not set the username is not added to the redirect_uri parameters

Reply:
No, condenser for example uses a search #hash for the username, better to use templating as I suggested in the issue

`redirectUri = DEFAULT_REDIRECT_URI.replace(usernameTemplateRegex, username)`

Reply:
What's the query thing btw? That might need to be merged with the templating as well

Reply:
`urlParameter` is here to handle any custom parameters passed in the query. This was here when we offered the possibility to define a custom redirect uri for any 3rd app willing to use the faucet for account creation. Since it's not used this parameter can be removed

Reply:
Ok, we want that for tracking id's very soon so make it a generic url template.

Spec:

  1. Values passed via query strings to first step are carried through the signup process
  1. Redirect uri is resolved via templating
      1. Variables in template look like: `{{variable_name}}`
      1. `username` is special and is populated by new username and not overridable by first step query strings

Example, with `DEFAULT_REDIRECT_URI` set to `https://example.com/{{username}}?foo={{bar}}` and an initial entry point to step1 with `?bar=man` and new username `baz` the redirect url should resolve to: `https://example.com/baz?foo=man`

Tests that should pass:

  * Template: `DEFAULT_REDIRECT_URI="http://localhost:1234/{{username}}/foo?u={{username}}&id={{id}}`
      * New username: `monika`
      * Step 1 query string: `?id=1234`
      * Result: `http://localhost:1234/monika/foo?u=monika&id=1234`
  * Template: `DEFAULT_REDIRECT_URI="https://example.com`
      * New username: `monika`
      * Step 1 query string: `?id=1234`
      * Result: `https://example.com`
  * Template: `DEFAULT_REDIRECT_URI="https://example.com#{{username}}`
      * New username: `monika`
      * Step 1 query string: `?username=hax`
      * Result: `https://example.com#monika`